### PR TITLE
misc: add helper tools for generate config headers

### PR DESCRIPTION
**Pull request by yuhaoth**
tls1.3 only mode need enable/disable many options.
With MBEDTLS_USER_CONFIG_FILE feature, we can add a header
file to easy the test.

This utils is to save local config change as a header file.
That can reduce the work load for generate new user config file. 

It is used for #5362 
## Status
**READY**

### CODE REVIEW

File: scripts/generate_config.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+# type: ignore
+
+"""Mbed TLS user configuration file generate.
```
Comment (gilles-peskine-arm) [inline scripts/generate_config.py:4]: Please document what this script does.

Without knowing what this is supposed to do, it's impossible to review.

**Comment by gilles-peskine-arm** [comment]
Sorry, I can't commit to reviewing this before mid-April.

File: scripts/generate_config.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+    def enable_arg(arg):
+        m = enable_arg_pair_regexp.match(arg)
+        assert m, arg
+        return True, m['name'], m['value']
+
+    def disable_arg(arg):
+        m = disable_arg_pair_regexp.match(arg)
+        assert m, arg
+        return False, m['name'], ''
```
Comment (gabor-mezei-arm) [inline scripts/generate_config.py:76]:
```suggestion
        return False, m['name'], ''

```
Add a new line for better separation.

File: scripts/generate_config.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+def get_changed(file_path):
+    """Get changed config in config.h"""
+    patch = PatchSet.from_string(
+                check_output(['git', 'diff', '--', file_path],
+                             universal_newlines=True))
+
+    def iter_lines():
+        for i in patch[0]:
+            yield from i.target_lines()
```
Comment (gabor-mezei-arm) [inline scripts/generate_config.py:47]:
```suggestion
            yield from i.target_lines()

```
Add a new line for better separation.

File: scripts/generate_config.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    config = ConfigFile(filename=args.input)
+    if args.over_write:
+        processed_settings = set()
+        for active, name, value in args.enable + args.disable:
+            assert name in config.settings, \
+                '{name} is not valid config'.format(**locals())
+            assert bool(config.settings[name].value) == bool(value), \
+                '{name}={value} is not valid config'.format(**locals())
+            assert name not in processed_settings, \
+                '{name} has been added in argument'.format(**locals())
```
Comment (gabor-mezei-arm) [inline scripts/generate_config.py:95]: The same checks done twice. Should be moved to a function.

**Review by gabor-mezei-arm — CHANGES_REQUESTED** [review]

**Review by gabor-mezei-arm — DISMISSED** [review]
LGTM

**Comment by yuhaoth** [comment]
rebased

**Review by gabor-mezei-arm — DISMISSED** [review]
LGTM

**Comment by yuhaoth** [comment]
Rebased to clear CI fail flag

**Review by gabor-mezei-arm — APPROVED** [review]
LGTM

**Comment by tom-cosgrove-arm** [comment]
Hi Jerry, I'm sorry, but after discussing with Dave and the other historical reviewers, we feel that there isn't a burning need for something like this, and anyway it would be better in `config.py` if anywhere, so I am closing this PR

**Pull request closed without merging**